### ISSUE TITLE
deposit: temporal metadata REST API

### ISSUE THREAD

**Issue by slint** [zenodo/zenodo#1741]
**Data model**

```python
{
   "dates": [
      {"start": "2019-03-31", "end": "2019-04-01", "type": "collected", "description": "A freeform textual description"},
   ],
}
```

**Deposit API**

- Granularity should be to the level of date: `YYYY-MM-DD`
- `type`: Controlled vocabulary from DataCite (to check if all or only some are made available)
- Validation: `start` should be before `end`, it should be possible to provide only `start` or only `end`, there should be at least one of them
- For single/exact values should be implemented by setting `start` and `end` to the same value

**Comment by ChiaraBi** [comment]
We will support the following types:  `Collected`, `Valid` and `Withdrawn` ( see https://schema.datacite.org/meta/kernel-4.2/doc/DataCite-MetadataKernel_v4.2.pdf )

**Comment by slint** [comment]
Closed via #1747
